app/models/edition.rb

```diff
@@ -30,6 +30,9 @@ class Edition < ActiveRecord::Base
   has_many :classification_featurings, inverse_of: :edition
   has_many :links_reports, as: :link_reportable
 
+  has_many :dependencies, class_name: 'EditionDependency', foreign_key: :edition_id, dependent: :destroy
```

Comment:
I think this could be simplified a little further (again following rails conventions) to:

```ruby
has_many :edition_dependencies, dependent: :destroy
has_many :contact_dependencies, through: :edition_dependencies, source: :dependable, source_type: 'Contact'
```